hazelcast/src/main/java/com/hazelcast/cache/impl/operation/CacheOperation.java

```diff
@@ -139,4 +138,29 @@ public final int getSyncBackupCount() {
     public final int getAsyncBackupCount() {
         return cache != null ? cache.getConfig().getAsyncBackupCount() : 0;
     }
+
+    protected final void publishWanUpdate(Data dataKey) {
+        if (!cache.isWanReplicationEnabled()) {
+            return;
+        }
+
+        CacheRecord record = cache.getRecord(dataKey);
```

Comment:
This means we will always do the lookup one more time for a WAN enabled cache, even though we already have the record in the subclass. Could this adversely affect performance? Can the method accept the value as the `MapOperation#publishWanUpdate` does?

Reply:
fixed.